internal/cmd/gocdk/provision.go

```diff
@@ -54,12 +59,164 @@ func registerProvisionCmd(ctx context.Context, pctx *processContext, rootCmd *co
+//                 "terraform init" again; currently we don't; see
+//                 https://github.com/google/go-cloud/issues/2291.
+// TODO(rvangent): Currently there are a bunch of default values for locals, so
+//                 "terraform apply" works fine. Add the ability to prompt the user
+//                 for things (e.g., GCP project ID), and use those values.
 func provisionAdd(ctx context.Context, pctx *processContext, biome, typ string) error {
-	return errors.New("not yet implemented")
+	pctx.Logf("Adding %q to %q...", typ, biome)
+
+	typeParts := strings.Split(typ, "/")
+	if len(typeParts) != 2 {
+		return xerrors.Errorf("provision add: %q is not a supported type; use 'gocdk provision list' to see available types", typ)
+	}
+
+	moduleDir, err := pctx.ModuleRoot(ctx)
+	if err != nil {
+		return xerrors.Errorf("provision add: %w", err)
+	}
+	dstPath := biomeDir(moduleDir, biome)
+
+	// TODO(rvangent): Use an explicit allowlist instead of just blindly reading "typ".
+	srcRoot := path.Join(provisionStaticRootDir, typ)
+	srcDir, err := static.Open(srcRoot)
+	if err != nil {
+		return xerrors.Errorf("provision add: %q is not a supported type; use 'gocdk provision list' to see available types", typ)
+	}
+	defer srcDir.Close()
+	srcInfos, err := srcDir.Readdir(-1)
+	if err != nil {
+		return xerrors.Errorf("provision add: couldn't read static resources: %w", err)
+	}
+	for _, srcInfo := range srcInfos {
+		name := srcInfo.Name()
+		if srcInfo.IsDir() {
+			return xerrors.Errorf("provision add: unexpected directory in static resources: %s", name)
+		}
+
+		srcFile, err := static.Open(path.Join(srcRoot, name))
+		if err != nil {
+			return xerrors.Errorf("provision add: couldn't open static resource %q: %w", name, err)
+		}
+		defer srcFile.Close()
+		srcBytes, err := ioutil.ReadAll(srcFile)
+		if err != nil {
+			return xerrors.Errorf("provision add: couldn't read static resource %q: %w", name, err)
+		}
+		// TODO(rvangent): There's a bunch of somewhat arcane logic here
+		// to figure out what to do with each file. There should be a config
+		// file for each directory that tells us what to do with each one.
+		// For example:
+		// -- By default files are just copied to the destination.
+		// -- A config setting that means "treat this as a template".
+		// -- A config setting means "insert into this file after this marker".
+		// -- A config setting means "append to this file".
+		// -- Options for insert/append to tell whether it should be skipped
+		//    (i.e., it's been added already), or whether to error.
+		// "demo add" and materializeTemplateDir could be updated to use this as well.
+		if strings.HasPrefix(name, "main_") {
+			if err := insertIntoFile(filepath.Join(dstPath, "main.tf"), "", srcBytes); err != nil {
+				return xerrors.Errorf("provision add: %w", err)
+			}
+		} else if strings.HasPrefix(name, "mainlocal_") {
+			const marker = "# DO NOT REMOVE: GO CDK LOCALS WILL BE INSERTED BELOW HERE"
```

Comment:
If I saw this, I'm not sure if I'd know whether it referred to "don't remove *this comment*" vs "don't remove whatever code is immediately below this" `DO NOT REMOVE THIS COMMENT` would be unambiguous.

Reply:
Done.